docs/jitserver_tuning.md

```diff
@@ -0,0 +1,125 @@
+
+## Session affinity
+
+For technical reasons a client JVM must use a single JITServer at a time. In a Kubernetes environment, where a JITServer service can be backed up by several server instances, you can satisfy this requirement by using session affinity. Note that if a server crashes (or gets terminated by the Kubernetes controller) the clients can connect to another server instance. This scenario imposes some performance penalty because the client-session caches that the server maintains need to be built anew. Below we show an example of a Kubernetes service definition that uses sessionAffinity:
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: jitserver
+spec:
+  type: ClusterIP
+selector:
+    app: jitserver
+  ports:
+    - protocol: TCP
+      port: 38400
+      targetPort: 38400
+  sessionAffinity: ClientIP
+  sessionAffinityConfig:
+    clientIP:
+      timeoutSeconds: 86400
+```
+
+## Resilience
+
+If the client JVM does not find a compatible server to connect to, compilations are performed locally, by the client itself. To account for the case where the server is temporarily unavailable (e.g, server crash followed by Kubernetes launching another server instance), from time to time the client retries to connect to a server at the indicated address and port. The retry mechanism uses an exponential back-off where the retry interval is doubled with each unsuccessful attempt.
+
+## Monitoring
+
+There are several ways to inspect the behavior of a JITServer instance, but all are based on the [OpenJ9 verbose logging facility](https://blog.openj9.org/2018/06/07/reading-verbose-jit-logs/). Note that if the name of the verbose log is not specified, the relevant information is printed to stderr.
+When you use the `-XX:+JITServerLogConnections` command line option, the server prints a message to the verbose log every time a new client JVM connects to it or disconnects from it. This is an easy way to determine that the clients are able to reach the server. E.g. of output:
+```
+#JITServer: t= 74232 A new client (clientUID=14692403771747196083) connected. Server allocated a new client session.
+#JITServer: t= 74282 A new client (clientUID=2599593246759846167) connected. Server allocated a new client session.
+#JITServer: t= 86281 Client (clientUID=14692403771747196083) disconnected. Client session deleted
+```
+
+The server has a heart-beat thread that periodically prints to the verbose log information related to the number of clients connected, the number of active compilation threads, the amount of CPU used, the amount of available memory and the number of times the internal server caches have been cleared. This last bit of information is important for diagnosing performance problems. The heart-beat information is enabled with the following option:
+
+    -Xjit:statisticsFrequency=<period-in-ms>
+
+E.g. of output:

```

Comment:
I suggest the abbreviation is unwanted here and on line 97 above; it will read better as, for example, 'Example output:'.